galsim/config/stamp.py

```diff
@@ -582,15 +588,16 @@ def _ParseDType(config, base):
     dtype = config.get('dtype', None)
     if isinstance(dtype, str):
         try:
-            gdict = globals().copy()
-            exec('import numpy', gdict)
-            exec('import numpy as np', gdict)
             try:
-                dtype = eval(dtype, gdict)
-            except Exception:
                 dtype = np.dtype(dtype).type
+            except Exception:
+                gdict = globals().copy()
+                exec('import numpy', gdict)
+                exec('import numpy as np', gdict)
+                dtype = eval(dtype, gdict)
         except Exception:
-            raise GalSimConfigValueError("dtype = %s is invalid."%dtype, Image.valid_dtypes)
+            raise GalSimConfigValueError(
+                "dtype = %s is invalid."%dtype, Image.valid_dtypes) from None
```

Comment:
This is unrelated to the rest of this PR.  When I was adding the from None here, I realized that the `dtype = np.dtype(dtype).type` option is faster, so should try that first rather than doing the whole exec bit.